### PR TITLE
Add new e2e test shopper my account pay order

### LINKED ISSUE #27884: Write new e2e test: Shopper / My Account (pay for order)
Write a new e2e test that covers Shopper / My Account flow.

The test should cover the following:
- shopper can pay for the order

### PR DESCRIPTION
**Pull request by veljkho**
### Changes proposed in this Pull Request:

Adding new core-test:  Shopper / My Account / (pay for order)

Closes #27884

### How to test the changes in this Pull Request:

1. npx wc-e2e test:e2e-dev ./tests/e2e/specs/front-end/test-my-account-pay-order.js

### Changelog entry

N/A

### CODE REVIEW

File: tests/e2e/env/config/default.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "url": "http://localhost:8084/",
+  "appName": "woocommerce_e2e",
```
Comment (rrennick) [inline tests/e2e/env/config/default.json:3]: @veljkho You need to revert this file.

**Review by zhongruige — COMMENTED** [review]
Thanks for adding in these tests @veljkho!

Ran the tests and everything worked well and passed for me locally:

```
  Customer can pay for his order through my account
    ✓ allows customer to pay for his order in my account (4287ms)
```

I left a couple of comments on minor text changes. There was just a couple of other things I wanted to mention:

* Looks like we have some merge conflicts on the following files (likely from merging in https://github.com/woocommerce/woocommerce/pull/28855)
  * `tests/e2e/core-tests/CHANGELOG.md`
  * `tests/e2e/core-tests/README.md`

* I think this file still needs to be reverted, based on @rrennick's comment here: https://github.com/woocommerce/woocommerce/pull/28818#discussion_r568104887

* I believe we want to avoid committing `package-lock.json` (@rrennick can probably help confirm my understanding here), though we will likely need a PR to push up the updated files so we don't get updates that causes unstaged changes in like this. `p1612797260319600-slack-C0E1AV8T0`

**Comment by rrennick** [comment]
> I believe we want to avoid committing package-lock.json

Yes, we should avoid updating `woocommerce/package-lock.json` unless the PR makes changes to `woocommerce/package.json`.

**Comment by veljkho** [comment]
Now package-lock.json is the same as on the master branch.

**Review by tammullen — COMMENTED** [review]
Thanks for this! Tests run great both locally and in CI
```
Customer can pay for their order through My Account
     ✓ allows customer to pay for their order in My Account (6446ms)
```

I agree with the suggested tweaks by @zhongruige 

Looks like we are still trying to commit `package-lock.json` and `default.json` and I don't believe we need to.

File: tests/e2e/config/default.json [hunk 2]
```diff
@@ -1,6 +1,5 @@
 {
   "url": "http://localhost:8084/",
-  "appName": "woocommerce_e2e",
```
Comment (rrennick) [inline tests/e2e/config/default.json:3]: @veljkho This wasn't the correct file. Can you revert it? I commented on `tests/e2e/env/config/default.json` vs `tests/e2e/config/default.json`. The `env` change was a result of the issue fixed in #29060.

File: tests/e2e/env/config/default/default.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "url": "http://localhost:8084/",
+  "appName": "woocommerce_e2e",
```
Comment (rrennick) [inline tests/e2e/env/config/default/default.json:3]: This file also needs to be reverted. The `env` version of the file should not have `appName` but the local config should have it.

**Review by zhongruige — DISMISSED** [review]
These changes look good! Thank you for adding in these tests @veljkho !

I see that the `package-lock.json` [change](https://github.com/woocommerce/woocommerce/pull/28818/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519) is just likely due to a minor spacing change, but I don't have specific concerns here:

Screen Shot 2021-03-03 at 8 41 02 AM

I believe this also addresses all of @rrennick's concerns as well, just tagging here to confirm.

**Comment by tammullen** [comment]
Looks like we are still trying to commit `package-lock.json` is this definitely needed? There are no corresponding changes to woocommerce/package.json.

**Comment by veljkho** [comment]
> Looks like we are still trying to commit `package-lock.json` is this definitely needed? There are no corresponding changes to `package.json`

I just fixed that. The difference was just a missing empty line at the top-bottom of the `package-lock.json` file.
After checks are ready we can consider merging.

**Review by tammullen — APPROVED** [review]
> I just fixed that.

Thanks! LGTM!

**Pull request merged**